cluster/examples/kubernetes/edgefs/operator.yaml

```diff
@@ -0,0 +1,222 @@
+  verbs: ["get", "list", "watch", "patch", "create", "update", "delete"]
+- apiGroups: ["extensions"]
+  resources: ["daemonsets"]
+  verbs: ["get", "list", "watch", "create", "update", "delete"]
+---
+# The cluster role for managing the Rook CRDs
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: rook-edgefs-global
+  labels:
+    operator: rook
+    storage-backend: edgefs
+rules:
+- apiGroups: [""]
+  # Pod access is needed for fencing
+  # Node access is needed for determining nodes where mons should run
+  resources: ["pods", "nodes", "nodes/proxy"]
+  verbs: ["get", "list", "watch", "update", "patch"]
+- apiGroups: [""]
+  # PVs and PVCs are managed by the Rook provisioner
+  resources: ["events", "persistentvolumes", "persistentvolumeclaims"]
+  verbs: ["get", "list", "watch", "patch", "create", "update", "delete"]
+- apiGroups: ["storage.k8s.io"]
+  resources: ["storageclasses"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["batch"]
+  resources: ["jobs"]
+  verbs: ["get", "list", "watch", "create", "update", "delete"]   
+- apiGroups: ["edgefs.rook.io"]
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["rook.io"]
+  resources: ["*"]
+  verbs: ["*"]
+---
+# The rook system service account used by the operator, agent, and discovery pods
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-edgefs-system
+  namespace: rook-edgefs-system
+  labels:
+    operator: rook
+    storage-backend: edgefs
+---
+# Grant the operator, agent, and discovery agents access to resources in the rook-ceph-system namespace
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-edgefs-system
+  namespace: rook-edgefs-system
+  labels:
+    operator: rook
+    storage-backend: edgefs
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: rook-edgefs-system
+subjects:
+- kind: ServiceAccount
+  name: rook-edgefs-system
+  namespace: rook-edgefs-system    
+---
+# Grant the rook system daemons cluster-wide access to manage the Rook CRDs, PVCs, and storage classes
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: rook-edgefs-global
+  namespace: rook-edgefs-system
+  labels:
+    operator: rook
+    storage-backend: edgefs
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-edgefs-global
+subjects:
+- kind: ServiceAccount
+  name: rook-edgefs-system
+  namespace: rook-edgefs-system
+---
+# The deployment for the rook operator
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: rook-edgefs-operator
+  namespace: rook-edgefs-system
+  labels:
+    operator: rook
+    storage-backend: edgefs
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: rook-edgefs-operator
+    spec:
+      serviceAccountName: rook-edgefs-system
+      containers:
+      - name: rook-edgefs-operator
+        image: edgefs/edgefs-operator:latest
```

Comment:
as discussed in the community call yesterday, we should update this image value to use `rook/edgefs` before going to master, so that folks can start using the officially published images after the Jenkins build publishes them.